contributor-docs/behaviors.md

```diff
@@ -0,0 +1,132 @@
+
+## Dependencies
+
+**External dependencies:** Third-party/external dependencies are not allowed, except under very strict circumstances. If you need to use an external dependency, open an issue and discuss with the team.
+
+**Internal dependencies:** Internal dependencies are fine and encouraged to reduce code duplication and maintain consistency. When taking an internal dependency, observe the following:
+
+* No circular dependencies
+* Components may depend on other components and component behaviors, but should avoid depending directly on generic behaviors.
+* Component behaviors may depend on other component behaviors and generic behaviors.
+* Generic behaviors may only depend on other generic behaviors.
+
+See below for a visualization of allowed dependencies.
+
+```
+               ____________
+              |            \*
++-------------V--------+   /
+|   Generic behaviors  |---
++----------------------+
+           ^   
+          *|     ___________
+           |    |           \*
++----------|----V-------+   /
+|  Component behaviors  |---
++-----------------------+
+         ^
+        1|    _____
+         |   |     \*
++--------|---V-+   /
+|  Components  |---
++--------------+
+
+```
+
+## Generic behaviors
+
+Generic behaviors provide functionality that is not specific to any single component or class of components. If you can't tell which component it should operate on, it's probably a generic behavior. Generic behaviors are _low-level_. Generic behaviors may only have dependencies on other generic behaviors.
+
+### Examples
+
+* `useProvidedRefOrCreate`
+* `usePosition`
+* `useClickAway`
+* `useTypeAhead`
+
+## Component behaviors
+
+Component behaviors specifically implement behaviors for components. Therefore, hooks that implement component behaviors should always return an object containing props objects to be spread across the elements that make up the component's JSX. While component behaviors do not necessarily have to be one-to-one, **a component may only use one component behavior**. Sometimes it makes sense to use the same component behavior on several component. For example: the Dialog, Popover, and PointerBox components may all use the `usePopover` component behavior.
+
+### Examples
+
+* `usePopover`
+* `useComboBox`
+* `useFilteredSearch`
+* `useDetails`
+
+## Implementation guidelines
+
+All behaviors should be implemented as [React Hooks](https://reactjs.org/docs/hooks-intro.html).
+
+Component behaviors should be relatively brief. A component behavior should build a component's experience through the combination of several configured generic behaviors.
+
+Example: The Dialog Component uses a `usePopover` behavior. The `usePopover` behavior in turn may call generic behaviors such as `usePosition` (to position the popover on the screen), `useClickAway` (to set up an event handler for clicking outside of the dialog), and `useFocusTrap` (to prevent focus from leaving the dialog while it is being shown).
```

Comment:
```suggestion
Example: The Dialog component uses a `usePopover` behavior. The `usePopover` behavior in turn may call generic behaviors such as `usePosition` (to position the popover on the screen), `useClickAway` (to set up an event handler for clicking outside of the dialog), and `useFocusTrap` (to prevent focus from leaving the dialog while it is being shown).
```